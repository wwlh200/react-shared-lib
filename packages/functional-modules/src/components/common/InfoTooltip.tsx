import * as React from 'react';

interface IPosition {
  position: string;
}

// tslint:disable-next-line:no-any
const InfoTooltip = (WrapComponent: any) =>
 class extends React.Component<IPosition> {
   constructor(props: IPosition) {
     super(props);
   }
   render() {
     return (
      <div className="info-tip">
          <WrapComponent />
      </div >
     );
   }
 };

export default InfoTooltip;
