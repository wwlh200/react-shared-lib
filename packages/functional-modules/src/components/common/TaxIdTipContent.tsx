import * as React from 'react';
import InfoTooltip from './InfoTooltip';

export class TaxIdTipContent extends React.PureComponent {
  render() {
    return (
      <React.Fragment>
        <span>test</span>
      </React.Fragment>
    );
  }
}

export default InfoTooltip(TaxIdTipContent);
