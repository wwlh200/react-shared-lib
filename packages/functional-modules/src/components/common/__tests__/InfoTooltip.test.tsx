import * as React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import InfoTooltip from '../InfoTooltip';
import TaxIdTipContent from '../TaxIdTipContent';

describe('InfoTooltip', () => {
  it('should render InfoTooltip', () => {
    const WrapComponent = InfoTooltip(TaxIdTipContent);
    const component = shallow(<WrapComponent position="a"/>);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
