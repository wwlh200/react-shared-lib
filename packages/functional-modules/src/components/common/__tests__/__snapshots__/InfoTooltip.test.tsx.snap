// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoTooltip should render InfoTooltip 1`] = `
<div
  className="info-tip"
>
  <Tooltip
    arrow={true}
    delay={150}
    linkClasses=""
    linkContent={
      <Icon
        altContent=""
        classes=""
        name="info"
        style={Object {}}
      />
    }
    position="a"
    positionOffset={5}
    reactive={true}
    theme="standard"
    trigger="click"
    width={260}
  >
    <class_1 />
  </Tooltip>
</div>
`;
